src/stores/Binding/Store.ts

```diff
@@ -1043,13 +1049,27 @@ const getInitialState = (
                     ? state.resourceConfigs[targetBindingUUID]
                     : null;
 
+                const targetResourceConfig =
+                    state.resourceConfigs[targetBindingUUID];
+
+                // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+                if (!targetResourceConfig) {
+                    logRocketEvent(
+                        CustomEvents.BINDINGS_RESOURCE_CONFIG_MISSING,
+                        {
+                            targetBindingUUID,
+                            targetCollection,
+                        }
+                    );
+
+                    return;
+                }
```

Comment:
This should not really happen and it was really hard to reproduce but putting this here and hoping we can see if this keeps happening.